channel/consensus_channel/follower_channel_test.go

```diff
@@ -0,0 +1,118 @@
+package consensus_channel
+
+import (
+	"errors"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/internal/testdata"
+	"github.com/statechannels/go-nitro/types"
+)
+
+var alice = types.Destination(common.HexToHash("0x0a"))
+var bob = types.Destination(common.HexToHash("0x0b"))
+var aBal = uint64(200)
+var bBal = uint64(300)
+var vAmount = uint64(5)
+var existingChannel = types.Destination{1}
+var targetChannel = types.Destination{2}
+
+// TODO these helpers and the helpers in leader_channel should be shared.
+func fp() state.FixedPart {
```

Comment:
Note that defining `fp` pollutes the global scope of the `consensus_channel` package: you can see below that `fp` is available inside the `follower_channel.go` file. (This is also true for the vars added on L14-L20).

![Screen Shot 2022-03-28 at 11 54 38 AM](https://user-images.githubusercontent.com/8432675/160467245-07c4828d-e848-4cec-a4ec-a5aaed2c21f2.png)

Reply:
FYI I don't know of a good solution to this problem -- this issue is what @NiloCK has been focusing on.

Reply:
I am a little confused about this, actually. [This post](https://stackoverflow.com/a/59060637) suggests that `fp` is _not_ compiled when building `go-nitro`. I'd infer from that that `fp` should not be available inside the `follower_channel.go` file.


I re-did the above experiment, with a different result:
![Screen Shot 2022-03-28 at 12 49 43 PM](https://user-images.githubusercontent.com/8432675/160475627-d29f7590-2d87-4e4b-bda9-6234007e6075.png)

Indeed, when I run `go test ./...` with the errant `fp` on the right, there's no problem.
```
➜  go-nitro git:(sign-next-proposal) ✗ go test ./... 
?       github.com/statechannels/go-nitro/abi   [no test files]
ok      github.com/statechannels/go-nitro/channel       (cached)
ok      github.com/statechannels/go-nitro/channel/consensus_channel     0.544s
ok      github.com/statechannels/go-nitro/channel/state (cached)
ok      github.com/statechannels/go-nitro/channel/state/outcome (cached)
?       github.com/statechannels/go-nitro/client        [no test files]
?       github.com/statechannels/go-nitro/client/engine [no test files]
ok      github.com/statechannels/go-nitro/client/engine/chainservice    (cached)
ok      github.com/statechannels/go-nitro/client/engine/messageservice  (cached)
ok      github.com/statechannels/go-nitro/client/engine/store   (cached)
ok      github.com/statechannels/go-nitro/client_test   0.499s
?       github.com/statechannels/go-nitro/crypto        [no test files]
ok      github.com/statechannels/go-nitro/protocols     (cached)
ok      github.com/statechannels/go-nitro/protocols/directdefund        (cached)
ok      github.com/statechannels/go-nitro/protocols/directfund  (cached)
ok      github.com/statechannels/go-nitro/protocols/virtualfund (cached)
ok      github.com/statechannels/go-nitro/types (cached)
```

But, `go build` fails:
```
➜  go-nitro git:(sign-next-proposal) go build ./...
# github.com/statechannels/go-nitro/channel/consensus_channel
channel/consensus_channel/follower_channel.go:59:9: undefined: fp
```

Reply:
This is a good point. Let's leave this to a future PR. I think it's an open question on what the best approach is.